core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1755,10 +1766,12 @@ public function generateRecordLabel(array $row, string $table = null, bool $prot
 				{
 					$args[$k] = $row[$v];
 				}
+
+				$args[$k] = (string) $args[$k] !== '' ? $args[$k] : '-';
 			}
 			else
 			{
-				$args[$k] = null;
+				$args[$k] = '-';
```

Comment:
This will have side-effects. Please revert.

Reply:
what side effects are you expecting? Since the same default value is applied above, e.g. if a column is `rgxp=date` but the value is empty?

Reply:
This is the status quo:

<img width="238" alt="" src="https://github.com/contao/contao/assets/1192057/1dd135ec-24a3-4119-bda3-430b2ba405d5">

If the first value is `null`, we e.g. replace the value with an icon.

But now the first value is no longer empty:

<img width="247" alt="" src="https://github.com/contao/contao/assets/1192057/ad95b6d9-51a7-4696-9d80-a14666e92b30">

This will have side-effects for sure!

Reply:
why would we not replace the icon if the value is not null? `tl_member:addIcon` always adds the icon at least.
Is this status quo the 4th callback argument?

Reply:
I have resolved the dash in 5265c250436d5b54e35fd4166a2e7821a1607742 since I noticed there are other places that rely on an empty value (e.g. the Picker widget).